publisher/internal/analytics/ethereum/convert.go

```diff
@@ -49,13 +49,26 @@ func convertTickToRatio(tick, token0Decimal, token1Decimal int) float64 {
 	return math.Pow(1.0001, float64(tick)) / math.Pow(10, float64(token1Decimal-token0Decimal))
 }
 
-func parseEventLogMessage(data []byte) EventLog {
-	var eLog EventLog
-	err := json.Unmarshal(data, &eLog)
+func parseEventLogMessage(data []byte) (WrappedEventLog, error) {
+	var eLog WrappedEventLog
+	err := json.Unmarshal(data, &eLog.Data)
 	if err != nil {
-		log.Println("--- ERROR: Encountered an error when unmarshaling EventLog:", err)
+		return WrappedEventLog{}, err
 	}
-	return eLog
+
+	if !eLog.hasTopics() {
+		return WrappedEventLog{}, fmt.Errorf("parsed event log has no topics.")
+	}
+
+	eLog.Instructions = otherEvent                    // Default instructions
+	for prefix, instruct := range eventInstructions { // Assign appropriate instructions
```

Comment:
The usage of `eventInstructions` looks like a factory method pattern. In other words, it would be much clearer to have a method/function that given a `eLog.Data.Topics[0]` (or simply `eLog`) will create an instruction.

This will not only remove the need for global variables(`eventInstructions`), but also will remove an implicit "singeton" property from event instructions(The `Operation` in this case).